src/test/java/io/goobox/sync/storj/mocks/FilesMock.java

```diff
@@ -161,6 +162,11 @@ public Path write(Path path, byte[] bytes, OpenOption... options) throws IOExcep
                 .filter(p -> p.startsWith(start));
     }
 
+    @Mock
+    public Path copy(Path source, Path target, CopyOption... options) throws IOException {
+        return Files.copy(source, target, options);
```

Comment:
Here you are just calling the method recursively - hence the StackOverflowError.

Instead, you should change the `files` map, if necessary. I guess in this case you can just leave it empty, i.e. doing nothing.